airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3NameTransformer.java

```diff
@@ -37,7 +37,15 @@ protected String applyDefaultCase(String input) {
 
   @Override
   public String getIdentifier(String name) {
-    return "_" + convertStreamName(name);
+    return checkFirsCharInStreamName(convertStreamName(name));
+  }
+
+  private String checkFirsCharInStreamName(String name) {
+    if (name.substring(0, 1).matches("\\d")) {
```

Comment:
would it be better if this should instead check if the first char is a valid first char (whitelist approach instead of blacklist)?

Reply:
of course I have updated s3 name transformer.